app/src/modules/insights/dashboard-utils/apply-data-to-panels.ts

```diff
@@ -0,0 +1,51 @@
+import { Panel } from '@directus/shared/types';
+
+export const applyDataToPanels = (panels: Panel[], incomingData: Record<any, any>) => {
+	if (!incomingData) return panels;
```

Comment:
If we expect null or undefined here as well, lets type it as such. I'm also assuming the incomingData object is most likely relying on strings as the key:

```suggestion
export const applyDataToPanels = (panels: Panel[], incomingData?: Record<string, any> | null) => {
	if (!incomingData) return panels;
```

Reply:
Also lets use regular functions for this export for consistencies sake:

```suggestion
export function applyDataToPanels(panels: Panel[], incomingData: Record<any, any>)  {
	if (!incomingData) return panels;
```